compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -640,6 +640,17 @@ impl<'a> Parser<'a> {
             }
         }
 
+        // extra info for `c"str"` before 2021 edition or `c "str"` in all editions
+        if self.prev_token.is_ident_named(Symbol::intern("c"))
```

Comment:
```suggestion
        if self.prev_token.is_ident_named(sym::c)
```

Reply:
You probably also want to check for `cr` (as in `cr""`, raw c-string literals). If I'm not mistaken “`sym::c && StrRaw`” only accounts for `c r""` (notice the space).